torch/fx/symbolic_trace.py

```diff
@@ -660,10 +686,43 @@ def symbolic_trace(root : Union[torch.nn.Module, Callable], concrete_args: Optio
     Given an ``nn.Module`` or function instance ``root``, this function will return a ``GraphModule``
     constructed by recording operations seen while tracing through ``root``.
 
+    ``concrete_args`` allows you to partially specialize your function, whether it's to remove control flow or data structures.
+
+    For example::
+
+        def f(a, b):
+            if b == True:
+                return a
+            else:
+                return a*2
+
+    FX can typically not trace through this due to the presence of control
+    flow. However, we can use `concrete_args` to specialize on the value of
+    `b` to trace through this.
+
+        f = fx.symbolic_trace(f, concrete_args={'b': False})
+        assert f(3, False)  == 6
+
+    Note that although you can still pass in different values of `b`, they will be ignored.
```

Comment:
One thing to note, I'm doing this not just for the pytree stuff, but also because I think the behavior is kind of awkward in other use cases as well.

For example, let's say you're currently using a function

```
def f(x, y):
     return g(x, y)
```
Now, we'd like to trace `g` but we need a concrete arg.  So elsewhere, we do `fx.symbolic_trace(g, concrete_args={'y': True})`. Now, depending on whether we've traced it or not, we need to change the call sites of `g`.

I think this specialization shouldn't be thought of as changing the function, but just "guaranteeing" certain properties about the original function.

I do agree that we should "properly" guarantee these properties by adding asserts, but unfortunately I'm not sure how we can do this properly for values that don't have a repr. If we're fine with just partial support, I can add that.

Reply:
I added support for ops in `base_types` and threw warnings for any other types.

I'm not sure what the right way to support arbitrary types here is.